src/transaction/log_compress.c

```diff
@@ -198,15 +198,18 @@ log_zip_realloc_if_needed (LOG_ZIP & log_zip, LOG_ZIP_SIZE_T new_size)
 
   if (new_size > 0 && new_size > log_zip.buf_size)
     {
-      log_zip.log_data = (char *) realloc (log_zip.log_data, new_size);
+      const LOG_ZIP_SIZE_T buf_size = LOG_ZIP_BUF_SIZE (new_size);
+      assert (buf_size <= LZ4_MAX_INPUT_SIZE);
```

Comment:
Are you sure this assert is required? Existing code only checks size before `LOG_ZIP_BUF_SIZE()` transformation

Reply:
@ribram
I thought that it makes sense, since `LOG_ZIP_BUF_SIZE` increases the input value. However, it is an unattainable corner case.